q2_diversity/_alpha/_visualizer.py

```diff
@@ -264,18 +264,19 @@ def _compute_rarefaction_data(feature_table, min_depth, max_depth, steps,
 
 
 def alpha_rarefaction(output_dir: str, table: biom.Table, max_depth: int,
-                      phylogeny: skbio.TreeNode=None, metric: str=None,
+                      phylogeny: skbio.TreeNode=None, metric: set=None,
                       metadata: qiime2.Metadata=None, min_depth: int=1,
                       steps: int=10, iterations: int=10) -> None:
+
     if metric is None:
         metrics = ['observed_otus', 'shannon']
         if phylogeny is not None:
             metrics.append('faith_pd')
     else:
```

Comment:
there should probably be a `elif not metrics:` because we don't have any dimension predicates to prevent the set from being empty.